halo2_proofs/src/dev.rs

```diff
@@ -315,6 +325,10 @@ impl<F: Field + Group> Assignment<F> for MockProver<F> {
         NR: Into<String>,
         N: FnOnce() -> NR,
     {
+        if !self.in_phase(FirstPhase) {
+            return;
+        }
```

Comment:
Why can't we `enter_region` with a different phase column? 
IIUC we should first ignore the `NotFirstPhaseColumn` and in the second and third etc... runs, indeed check it.

Reply:
Hmmm that makes sense..